### ISSUE TITLE
Invalid evaluation of futures

### ISSUE THREAD

**Issue by wytseoortwijn** [utwente-fmt/vercors#103]
The following PVL program containing a future should not verify, but it does however:

```java
class Future {
	int val;
	
	modifies val;
	ensures val == \old(val) + n;
	process incr(int n); 

	modifies val;
	ensures val == \old(val) * n;
	process mult(int n);
	
	ensures val == (\old(val) + n) * n || val == (\old(val) * n) * n;
	process P(int n) = incr(n) || mult(n);
}
```

**Comment by wytseoortwijn** [comment]
The following line states that the `--check-defined` flag should be used to check whether a process satisfies its contract:

https://github.com/utwente-fmt/vercors/blob/41c8da9bd4b41a62aac76ea694f5ac6fff069576/vercors/src/main/java/vct/main/Main.java#L732

However, this flag seems to be broken on this example and gives a `java.lang.NullPointerException`. I'll make a separate issue to resolve this.

**Comment by wytseoortwijn** [comment]
See #105

**Comment by wytseoortwijn** [comment]
Verifying the following snippet with `--check-defined` produces a failing verdict after #138.

```java
class Future {
	int val;
	
	modifies val;
	ensures val == \old(val) + n;
	process incr(int n); 

	modifies val;
	ensures val == \old(val) * n;
	process mult(int n);
	
	modifies val;
	ensures val == (\old(val) + n) * n || val == (\old(val) * n) * n;
	process P(int n) = incr(n) || mult(n);
}
```

(notice the extra `modifies` clause added to the `P` process, which is needed to generate all the required permissions).
